src/sys/windows_impl.rs

```diff
@@ -291,3 +293,121 @@ fn round_up(num_to_round: i64, multiple: i64) -> i64 {
+/// that retrieves the volume mount point where the specified path is mounted.
+fn get_volume_path(path: impl AsRef<Path>) -> io::Result<Vec<u16>> {
+    let path_wide: Vec<u16> = path
+        .as_ref()
+        .as_os_str()
+        .encode_wide()
+        .chain(Some(0))
+        .collect();
+    let mut volume_name_buffer = vec![0u16; MAX_PATH as usize];
+
+    unsafe { GetVolumePathNameW(PCWSTR(path_wide.as_ptr()), volume_name_buffer.as_mut()) }?;
+
+    if let Some(pos) = volume_name_buffer.iter().position(|&c| c == 0) {
+        volume_name_buffer.truncate(pos);
+    }
+
+    Ok(volume_name_buffer)
+}
+
+/// A wrapper function for
+/// [GetVolumeNameForVolumeMountPointW](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumenameforvolumemountpointw)
+/// that retrieves a volume GUID path for the volume that is associated with the specified volume
+/// mount point (drive letter, volume GUID path, or mounted folder).
+fn get_volume_guid_path(volume_path_w: &[u16]) -> io::Result<Vec<u16>> {
+    let mut volume_guid_path = vec![0u16; 50usize];
+    unsafe {
+        GetVolumeNameForVolumeMountPointW(PCWSTR(volume_path_w.as_ptr()), volume_guid_path.as_mut())
+    }?;
+
+    if let Some(pos) = volume_guid_path.iter().position(|&c| c == 0) {
+        volume_guid_path.truncate(pos);
+    }
+
+    Ok(volume_guid_path)
+}
+
+/// A wrapper function for
+/// [GetVolumeInformationW](https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getvolumeinformationw)
+/// that returns `FileSystemFlags`.
+fn get_volume_flags(volume_path_w: &[u16]) -> io::Result<u32> {
+    let mut file_system_flags = 0u32;
+
+    unsafe {
+        GetVolumeInformationW(
+            PCWSTR(volume_path_w.as_ptr()),
+            None,
+            None,
+            None,
+            Some(&mut file_system_flags as *mut _),
+            None,
+        )
+    }?;
+
+    Ok(file_system_flags)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_get_volume_path_is_same() -> io::Result<()> {
+        let src_volume_path = get_volume_path("./src")?;
+        let tests_volume_path = get_volume_path("./tests")?;
+        assert_eq!(src_volume_path, tests_volume_path);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_get_volume_guid() -> io::Result<()> {
+        let volume_path = get_volume_path(".")?;
+
+        let re = regex::Regex::new(r"\\\\\?\\Volume\{.{8}-.{4}-.{4}-.{4}-.{12}\}\\").unwrap();
+        let volume_guid = get_volume_guid_path(&volume_path)?;
+        let volume_guid = String::from_utf16(&volume_guid).unwrap();
+        assert!(re.is_match(&volume_guid));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_get_volume_flags() -> io::Result<()> {
+        let volume_path = get_volume_path(".")?;
+        let volume_flags = get_volume_flags(&volume_path)?;
+        assert!(volume_flags > 0);
+        Ok(())
+    }
+}
```

Comment:
For testing, I think we can create ci-only test.

On the CI we could setup dev-drive and pass the newly setup dev-drive path as an env.

https://github.com/marketplace/actions/setup-dev-drive

If that env is present, then we run the test, otherwise we skip it.

Using dev-drive, we can test that `check_reflink_support` returns `NotSupported` on Win for different drive.

If we want to test that for a fs that does not support reflink, it will also return `NotSupported` if src/dst within same fs, then we could setup WSL

It's a good edge case that might be encountered on dev machine.

https://github.com/marketplace/actions/setup-wsl

Reply:
I'll look into it

Reply:
For the WSL part, if you think it's too complex you can skip it.

But I think the dev-drive is pretty easy (try setting up one myself before) and you can use it to test many functions here.

Reply:
After overcoming some challenges, I've managed to add all the tests that I believe are appropriate for this functionality. I hope you find them helpful :)

I had to hide the new tests behind `#[ignore]` because they depend on volumes set up in the pipeline and will fail if you try to run them locally.